drivers/display/ssd1673.c

```diff
@@ -0,0 +1,512 @@
+	struct spi_buf buf = {.buf = &cmd, .len = 1};
+	struct spi_buf_set buf_set = {.buffers = &buf, .count = 1};
+	bool update = true;
+
+	if (!data || !len) {
+		return -1;
+	}
+
+	ssd1673_busy_wait(driver);
+	memset(dummy_page, 0xff, sizeof(dummy_page));
+
+	switch (driver->scan_mode) {
+	case SSD1673_DATA_ENTRY_XIYDY:
+		if (ssd1673_set_ram_param(driver,
+					  SSD1673_PANEL_FIRST_PAGE,
+					  SSD1673_PANEL_LAST_PAGE + 1,
+					  SSD1673_PANEL_LAST_GATE,
+					  SSD1673_PANEL_FIRST_GATE)) {
+			return -1;
+		}
+
+		if (ssd1673_set_ram_ptr(driver,
+					SSD1673_PANEL_FIRST_PAGE,
+					SSD1673_PANEL_LAST_GATE)) {
+			return -1;
+		}
+
+		break;
+
+	case SSD1673_DATA_ENTRY_XDYIY:
+		if (ssd1673_set_ram_param(driver,
+					  SSD1673_PANEL_LAST_PAGE + 1,
+					  SSD1673_PANEL_FIRST_PAGE,
+					  SSD1673_PANEL_FIRST_GATE,
+					  SSD1673_PANEL_LAST_GATE)) {
+			return -1;
+		}
+
+		if (ssd1673_set_ram_ptr(driver,
+					SSD1673_PANEL_LAST_PAGE + 1,
+					SSD1673_PANEL_FIRST_GATE)) {
+			return -1;
+		}
+
+		break;
+	default:
+		return -1;
+	}
+
+	gpio_pin_write(driver->dc, CONFIG_SSD1673_DC_PIN, 0);
+	if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
+		return -1;
+	}
+
+	gpio_pin_write(driver->dc, CONFIG_SSD1673_DC_PIN, 1);
+	/* clear unusable page */
+	if (driver->scan_mode == SSD1673_DATA_ENTRY_XDYIY) {
+		buf.buf = dummy_page;
+		buf.len = sizeof(dummy_page);
+		if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
+			return -1;
+		}
+	}
+
+	buf.buf = (u8_t *)data;
+	buf.len = len;
+	if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
+		return -1;
+	}
+
+	/* clear unusable page */
+	if (driver->scan_mode == SSD1673_DATA_ENTRY_XIYDY) {
+		buf.buf = dummy_page;
+		buf.len = sizeof(dummy_page);
+		if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
+			return -1;
+		}
+	}
+
+
+	if (update) {
+		if (driver->contrast) {
+			return ssd1673_update_display(dev, true);
+		}
+		return ssd1673_update_display(dev, false);
+	}
+
+	return 0;
+}
+
+static int ssd1673_read_bitmap(const struct device *dev, const u16_t x,
+			const u16_t y, const u16_t w, const u16_t h,
+			const u16_t p, void *data)
+{
+	SYS_LOG_ERR("not supported");
+	return -ENOTSUP;
+}
+
+static void *ssd1673_get_framebuffer(const struct device *dev)
+{
+	SYS_LOG_ERR("not supported");
+	return NULL;
+}
+
+static int ssd1673_set_brightness(const struct device *dev,
+				  const u8_t brightness)
+{
+	SYS_LOG_WRN("not supported");
+	return -ENOTSUP;
+}
+
+int ssd1673_set_contrast(const struct device *dev, u8_t contrast)
+{
+	struct ssd1673_data *driver = dev->driver_data;
+
+	driver->contrast = contrast;
+
+	return 0;
+}
+
+static void ssd1673_get_capabilities(const struct device *dev,
+				     struct display_capabilities *caps)
+{
+	memset(caps, 0, sizeof(struct display_capabilities));
+	caps->x_resolution = EPD_PANEL_WIDTH;
+	caps->y_resolution = EPD_PANEL_HEIGHT;
+	caps->supported_pixel_formats = PIXEL_FORMAT_MONO10;
```

Comment:
Is this correct, PIXEL_FORMAT_MONO10 defines black as 1 and white as 0 but normal controller operation is the opposite.

Reply:
for SSD1673 controller 1 is black, e.g. 0x81 ->
```
[x]
[ ]
[ ]
[ ]
[ ]
[ ]
[ ] 
[x]
```